core/src/main/java/org/apache/accumulo/core/conf/AccumuloConfiguration.java

```diff
@@ -512,6 +512,12 @@ public T derive() {
       if (rc == null || rc.count != uc) {
         T newObj = converter.apply(AccumuloConfiguration.this);
 
+        if (newObj == null) {
+          // The converter should not return a null value, if it does then
+          // use the previous value.
+          return rc.obj;
```

Comment:
The reason for #4423 was that the classloader would have a transient error and return a null object when attempting to load the class. 

The Deriver would then cache this null object and return it even though the classloader had since recovered from its issue.

However, since the property value was technically still "correct" the user would need to either A. delete it, or B. set it to an invalid value in order to trigger the Deriver to reload the object.

So, I don't think that returning the previous object would fix that error condition.